evm_arithmetization/src/cpu/kernel/asm/mpt/linked_list/linked_list.asm

```diff
@@ -507,41 +502,31 @@ next_node_ok_with_value:
     // stack: new_ptr + 1, next_ptr, addr_key, key, value, retdest
     // Append the value to `TrieDataSegment` and write the resulting payload_ptr.
     %increment
-    DUP1
-    %get_trie_data_size
-    // stack: new_payload_ptr, new_ptr+2, new_ptr+2, next_ptr, addr_key, key, value, retdest
-    %stack (new_payload_ptr, new_payload_ptr_ptr, new_payload_ptr_ptr, next_ptr, addr_key, key, value, retdest)
-        -> (value, new_payload_ptr, new_payload_ptr_ptr, new_payload_ptr_ptr, next_ptr, new_payload_ptr, retdest)
-    %append_to_trie_data
-    MSTORE_GENERAL
-
-    // stack: new_ptr + 2, next_ptr, new_payload_ptr, retdest
-    // Store the payload ptr copy
-    %increment
-    DUP1
-    DUP4
-    %clone_slot
-    MSTORE_GENERAL
-    // stack: new_ptr + 3, next_ptr, new_payload_ptr, retdest
+    DUP1 %increment
+    // stack: new_ptr+3, new_payload_ptr_ptr, next_ptr, addr_key, key, value, retdest
```

Comment:
The variable names with `payload_ptr` are out of date, since we're now dealing with values directly.